presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java

```diff
@@ -170,6 +170,9 @@ public static HiveStorageFormat getHiveStorageFormat(Map<String, Object> tablePr
         if (bucketCount < 0) {
             throw new PrestoException(INVALID_TABLE_PROPERTY, format("%s must be greater than zero", BUCKET_COUNT_PROPERTY));
         }
+        if (bucketCount > 32_000) {
```

Comment:
Why we want to fail such queries? -- we can still create such table right ? -- just Presto cannot insert into such table. 

Also note this upper limit depends on the cluster configs. For example once our default hash partitions becomes 666, we can insert into 65536 buckets table. 32000 is not a hard limit.

Reply:
This is just a fail-fast. Anyhow we cannot reliably write / read tables with higher bucket count. If some other engine can handle that - let the table be created by the other engine.